setup.py

```diff
@@ -21,29 +21,61 @@
 import os
 import platform
 
-from setuptools import setup, Extension, Feature
+
+from distutils.errors import CCompilerError
+from distutils.errors import DistutilsExecError
+from distutils.errors import DistutilsPlatformError
+
+from setuptools import Extension
+from setuptools.command.build_ext import build_ext
+from setuptools import setup
+from setuptools import Feature
+
+class optional_build_ext(build_ext):
+    """This class subclasses build_ext and allows
+       the building of C extensions to fail.
+    """
+    def run(self):
+        try:
+            build_ext.run(self)
+        except DistutilsPlatformError as e:
+            self._unavailable(e)
+
+    def build_extension(self, ext):
+        try:
+            build_ext.build_extension(self, ext)
+        except (CCompilerError, DistutilsExecError, OSError) as e:
+            self._unavailable(e)
+
+    def _unavailable(self, e):
+        print('*' * 80)
+        print("""WARNING:
+        An optional code optimization (C extension) could not be compiled.
+        Optimizations for this package will not be available!""")
+        print()
+        print(e)
+        print('*' * 80)
+
 
 def read(*rnames):
     with open(os.path.join(os.path.dirname(__file__), *rnames)) as f:
         return f.read()
 
 Cwrapper = Feature(
     "C wrapper",
-    standard = True,
+    standard=True,
     headers=[os.path.join('src', 'zope', 'proxy', 'proxy.h')],
-    ext_modules=[Extension("zope.proxy._zope_proxy_proxy",
-                            [os.path.join('src', 'zope', 'proxy',
-                                        "_zope_proxy_proxy.c")
-                            ],
-                            extra_compile_args=['-g']),
-                ],
+    ext_modules=[
+        Extension(
+            "zope.proxy._zope_proxy_proxy",
+            [os.path.join('src', 'zope', 'proxy', "_zope_proxy_proxy.c")],
```

Comment:
Are you intentionally dropping the `extra_compile_args=['-g']`?

Reply:
I am, yes.

The use of extra_compile_args is inconsistent in zopefoundation projects. As far as I had seen, only this one and zope.hookable did so. zope.security, btrees, zope.interface all did not.

Because the point of this extension is moar speedz, it seems to make sense not to distribute it in a way that can compromise that goal. For local development, it's easy enough for a developer to set CFLAGS.